clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/InAppResourceProvider.kt

```diff
@@ -4,83 +4,97 @@ import android.content.Context
 import android.graphics.Bitmap
 import android.graphics.BitmapFactory
 import com.clevertap.android.sdk.ILogger
+import com.clevertap.android.sdk.inapp.images.memory.FileMemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.GifMemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.ImageMemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.MemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.MemoryCreator
+import com.clevertap.android.sdk.inapp.images.memory.TRANSFORM_TO_BITMAP
+import com.clevertap.android.sdk.inapp.images.memory.TRANSFORM_TO_BYTEARRAY
+import com.clevertap.android.sdk.inapp.images.memory.TRANSFORM_TO_FILE
 import com.clevertap.android.sdk.network.DownloadedBitmap
 import com.clevertap.android.sdk.utils.CTCaches
 import java.io.ByteArrayOutputStream
 import java.io.File
 
-internal class InAppResourceProvider constructor(
+internal class InAppResourceProvider(
     private val images: File,
     private val gifs: File,
+    private val allFileTypesDir: File,
     private val logger: ILogger? = null,
-    private val ctCaches: CTCaches = CTCaches.instance(logger = logger),
-    private val fileToBitmap : (file: File?) -> Bitmap? = { file ->
-        if (file != null && file.hasValidBitmap()) {
-            BitmapFactory.decodeFile(file.absolutePath)
-        } else {
-            null
-        }
-    },
-    private val fileToBytes: (file: File?) -> ByteArray? = { file ->
-        file?.readBytes()
-    },
     private val inAppRemoteSource: InAppImageFetchApiContract = InAppImageFetchApi()
 ) {
+
+    private var ctCaches: CTCaches = CTCaches.instance(
```

Comment:
names to call args and formatting

Reply:
fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/24ae903b1e807dc71671dda11097b5cbb99dacf4